src/plugins/imagery/components/ImageryViewLayout.vue

```diff
@@ -199,13 +237,20 @@ export default {
             }
 
             return result;
+        },
+        shouldDisplayCompass() {
+            return this.focusedImage !== undefined
+                && this.focusedImageNaturalAspectRatio !== undefined
+                && this.imageContainerWidth !== undefined
+                && this.imageContainerHeight !== undefined;
```

Comment:
We could add these variables into an array and do: VARIABLES.all(variable => variable !== undefined), so we are not repeating logic.